src/notification.ts

```diff
@@ -91,31 +88,21 @@ export class BaseNotification {
     this.gateway = gateway;
 
     // Load fetch:
-    this.fetch = crossFetch;
-    this.setSessionFetch();
-
-    this.fetchLoaded = false;
-    this.fetchLoader = new Promise<void>((resolve) => {
-      if (fetchFn) {
-        this.setSessionFetch(fetchFn);
-        resolve();
-      } else {
-        // Attempt to load the fetch function from the default session if no fetchFn was passed in.
-        BaseNotification.getDefaultSessionFetch()
-          .then((defaultFetchFn) => {
-            if (defaultFetchFn) {
-              this.setSessionFetch(defaultFetchFn);
-            }
-          })
-          .finally(() => {
-            resolve();
-          });
-      }
-    }).then(() => {
-      this.fetchLoaded = true;
-    });
+    if (fetchFn) {
+      this.fetch = fetchFn;
+    } else {
+      // Attempt to load the fetch function from the default session if no fetchFn was passed in
+      this.fetchLoader = BaseNotification.getDefaultSessionFetch()
```

Comment:
A note for posterity - an even simpler option remove `fetchLoader`; and give `this.fetch` type of `MaybePromise<typeof fetch>`, each method where it is used can just then run `const fetch = await this.fetch` and then replace this else statement with the following.

```ts
this.fetch = BaseNotification.getDefaultSessionFetch()
 .then(defaultFetchFn => defaultFetchFn || crossFetch)
 .catch(() => crossFetch);
```

I haven't done this due to the downside that this is this then forces the library to wait an extra tick in lines like line 205.

Reply:
Yeah, I'd probably be inclined to avoid making `this.fetch` a promise that when invoked would return a promise. I'd also avoid having fetch be optional, as then you'll need explicit checks for "is fetch defined" before invoking it.

maybe we can just define a fetch method that does the loading if necessary on first invocation and has the exact fetch API, that might simplify all this.